extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -0,0 +1,383 @@
+  }
+
+  /**
+   * Handles swipeY events, detected by the touch events listeners.
+   * @param {{event: !Event, data: !Object}} gesture
+   * @private
+   */
+  onSwipeY_(gesture) {
+    const {data} = gesture;
+
+    if (this.ignoreCurrentSwipeYGesture_ === true) {
+      this.ignoreCurrentSwipeYGesture_ = !data.last;
+      return;
+    }
+
+    const {deltaY, swipingUp} = data;
+
+    // If the attachment is open, figure out if the user is trying to scroll the
+    // content, or actually close the attachment.
+    if (this.state_ === AttachmentState.OPEN) {
+      const isContentSwipe = this.isAttachmentContentDescendant_(
+          dev().assertElement(gesture.event.target));
+
+      // If user is swiping up, exit so the event bubbles up and maybe scrolls
+      // the attachment content.
+      // If user is swiping down and scrollTop is above zero, exit and let the
+      // user scroll the content.
+      // If user is swiping down and scrollTop is zero, don't exit and start
+      // dragging/closing the attachment.
+      if ((isContentSwipe && deltaY < 0) ||
+          (isContentSwipe && deltaY > 0 &&
+              this.containerEl_./*OK*/scrollTop > 0)) {
+        this.ignoreCurrentSwipeYGesture_ = true;
+        return;
+      }
+    }
+
+    gesture.event.preventDefault();
+
+    if (data.last === true) {
+      if (this.state_ === AttachmentState.DRAGGING_TO_CLOSE) {
+        (!swipingUp && deltaY > TOGGLE_THRESHOLD_PX) ?
+          this.close_() :
+          this.open();
+      }
+
+      if (this.state_ === AttachmentState.DRAGGING_TO_OPEN) {
+        (swipingUp && -deltaY > TOGGLE_THRESHOLD_PX) ?
+          this.open() :
+          this.close_();
+      }
+      return;
+    }
+
+    this.drag_(deltaY);
+  }
+
+  /**
+   * Whether the element is a descendant of attachment-content.
+   * @param {!Element} element
+   * @return {boolean}
+   * @private
+   */
+  isAttachmentContentDescendant_(element) {
+    return !!closest(element, el => {
+      return el.classList.contains('i-amphtml-story-page-attachment-content');
+    }, /* opt_stopAt */ this.element);
+  }
+
+  /**
+   * Drags the attachment on the screen upon user interaction.
+   * @param {number} deltaY
+   * @private
+   */
+  drag_(deltaY) {
+    let translate;
+
+    switch (this.state_) {
+      case AttachmentState.CLOSED:
+      case AttachmentState.DRAGGING_TO_OPEN:
+        if (deltaY > 0) {
+          return;
+        }
+        this.state_ = AttachmentState.DRAGGING_TO_OPEN;
+        translate = `translate3d(0, calc(100% + ${deltaY}px), 0)`;
+        break;
+      case AttachmentState.OPEN:
+      case AttachmentState.DRAGGING_TO_CLOSE:
+        if (deltaY < 0) {
+          return;
+        }
+        this.state_ = AttachmentState.DRAGGING_TO_CLOSE;
+        translate = `translate3d(0, ${deltaY}px, 0)`;
+        break;
+    }
+
+    this.mutateElement(() => {
+      this.element.classList.add('i-amphtml-story-page-attachment-dragging');
```

Comment:
What does this class do? I can't seem to find it on the css.

Reply:
Good catch! Looks like I've been able to do without it :))